### PR TITLE
Multi-sig wallet on individual bounties

### PR DESCRIPTION
**Pull request by Devorein**

### CODE REVIEW

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx [hunk 1]
```diff
@@ -89,16 +102,71 @@ async function switchActiveNetwork (chainId: number) {
...
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const { data: safeInfos } = useSWR(
+    (signer && account && chainId) ? `/connected-gnosis-safes/${account}` : null,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    () => getSafesForAddress({ signer: signer!, chainId: chainId!, address: account! })
```
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:159]: i think you want to use getSafesForAddresses()

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx [hunk 2]
```diff
@@ -89,16 +102,71 @@ async function switchActiveNetwork (chainId: number) {
...
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const { data: safeInfos } = useSWR(
+    (signer && account && chainId) ? `/connected-gnosis-safes/${account}` : null,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:158]: it is a bad practice to disable eslint rules. Please fix the code instead (that's why we have the lint rules right?)

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx [hunk 3]
```diff
@@ -89,16 +102,71 @@ async function switchActiveNetwork (chainId: number) {
...
+    () => getSafesForAddress({ signer: signer!, chainId: chainId!, address: account! })
+  );
+
+  const safeDataRecord = useMemo(() => {
```
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:162]: Please don't use useMemo just to save on a reduce over a few items.

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx (hunk 1 above)
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:159]: we should use chainIdToUse, which is the chain we'd need to make a payment on. When the user chooses to make a payment, we can ask them to switch their wallet to that network (thats what we do for normal payments)

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx [hunk 4]
```diff
@@ -89,16 +102,71 @@ async function switchActiveNetwork (chainId: number) {
...
+  const open = Boolean(anchorEl);
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const { data: safeInfos } = useSWR(
+    (signer && account && chainId) ? `/connected-gnosis-safes/${account}` : null,
```
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:157]: You should also include chainId into the SWR  'key' i think

File: components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx (hunk 3 above)
Comment (Devorein) [reply components/[pageId]/DocumentPage/components/BountyProperties/components/BountyApplicantsTable/BountyPaymentButton.tsx:162]: yeah, my bad 😅 I always forget not to use `useMemo` when generating a record. Will remove it

**Review by mattcasey — APPROVED** [review]

**Pull request merged**